pkgs/by-name/ar/ares/package.nix

```
@@ -2,58 +2,81 @@
   lib,
   SDL2,
   alsa-lib,
-  autoPatchelfHook,
+  apple-sdk_12,
+  cmake,
   fetchFromGitHub,
   gtk3,
   gtksourceview3,
+  gnumake,
   libGL,
   libGLU,
   libX11,
   libXv,
   libao,
-  libicns,
   libpulseaudio,
   librashader,
+  ninja,
+  moltenvk,
   openal,
   pkg-config,
   stdenv,
+  stdenvNoCC,
   udev,
   vulkan-loader,
-  which,
   wrapGAppsHook3,
+  zlib,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "ares";
-  version = "141";
+  version = "142";
 
   src = fetchFromGitHub {
     owner = "ares-emulator";
     repo = "ares";
-    rev = "v${finalAttrs.version}";
-    hash = "sha256-iNcoNdGw/DfYc9tsOGsPYoZLhVwNzJe8bVotx6Rl0j4=";
+    rev = "v142";
+    hash = "sha256-D/6r4f6dAIsP9JglAmzxlCwV5NoOuw2bZMZ2T1oJ9Jc";
   };
 
-  patches = [
-    ./patches/001-dont-rebuild-on-install.patch
-  ];
+  slang-shaders = stdenvNoCC.mkDerivation {
+    name = "slang-shaders";
+
+    src = finalAttrs.src;
+    sourceRoot = "source/thirdparty/slang-shaders";
+
+    postPatch = ''
+      substituteInPlace Makefile \
+        --replace-fail /usr '$(out)'
+    '';
+
+    buildInputs = [
+      gnumake
+    ];
```

Is `make` actually used at run‐time? Otherwise I’d suggest to put this in `nativeBuildInputs`:

```suggestion
    nativeBuildInputs = [
      gnumake
    ];
```

---

I'm impressed I keep messing this up. Something about the naming just alwys gets me - nativeBuildInputs? Native to what? The target? Nope.

---

The naming is messy, if you look at <https://nixos.org/manual/nixpkgs/unstable/#ssec-stdenv-dependencies-propagated>. Perhaps they should have been re‐named `depsBuildHost` and `depsHostTarget` for consistency.

But then you’d still have the confusion that _host_ and _target_ cause.